File: apps/backend/internal/models/todo.go
package models

import (
	"time"

	"github.com/google/uuid"
)

// Todo represents a todo item in the database and API.
type Todo struct {
	ID        uuid.UUID `json:"id"`
	Title     string    `json:"title"`
	Completed bool      `json:"completed"`
	CreatedAt time.Time `json:"created_at"`
	UpdatedAt time.Time `json:"updated_at"`
}

// CreateTodoInput is the input for creating a todo.
type CreateTodoInput struct {
	Title     string `json:"title" binding:"required,min=1"`
	Completed bool   `json:"completed"`
}

// UpdateTodoInput is the input for updating a todo (all fields optional).
type UpdateTodoInput struct {
	Title     *string `json:"title"`
	Completed *bool   `json:"completed"`
}

File: packages/types/src/todo.ts
import { z } from "zod";

/** Todo item schema (shared shape for API and OpenAPI) */
export const TodoSchema = z.object({
  id: z.string().uuid(),
  title: z.string().min(1),
  completed: z.boolean(),
  created_at: z.string().datetime(),
  updated_at: z.string().datetime(),
});

/** Body schema for creating a todo */
export const CreateTodoBodySchema = z.object({
  title: z.string().min(1),
  completed: z.boolean().optional(),
});

/** Body schema for updating a todo (all fields optional) */
export const UpdateTodoBodySchema = z.object({
  title: z.string().min(1).optional(),
  completed: z.boolean().optional(),
});

export type Todo = z.infer<typeof TodoSchema>;
export type CreateTodoBody = z.infer<typeof CreateTodoBodySchema>;
export type UpdateTodoBody = z.infer<typeof UpdateTodoBodySchema>;
